src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,109 @@
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+
+                    return this._googlePayPaymentProcessor.displayWallet()
+                        .then(paymentData => this._paymentInstrumentSelected(paymentData))
+                        .then(() => onPaymentSelect())
+                        .then(() => this._getPayment())
+                        .catch(error => {
+                            if (error.statusCode !== 'CANCELED') {
+                                onError(error);
+                            }
+                        });
+                }
+
+            })
+            .then(() => this._convert_token(testMode, checkoutcomkey))
+            .then(payment => {
+                const paymentData: Payment = payment as Payment;
+
+                return this._store.dispatch(this._orderActionCreator.submitOrder({useStoreCredit: payload.useStoreCredit}, options))
+                    .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(paymentData)));
+            });
+    }
+
+    async requestCheckoutcomTokenize(testMode: boolean | undefined, checkoutcomKey: string, data = {}): Promise<CheckoutcomToken> {
+
+        let url = '';
+        if (testMode) {
+            url = this.TEST_URL;
+        } else {
+            url = this.LIVE_URL;
+        }
+
+        const response = await fetch(url, {
+            method: 'POST',
+            mode: 'cors',
+            cache: 'no-cache',
+            credentials: 'omit',
+            headers: {
+                'Content-Type': ContentType.Json,
+                Authorization: checkoutcomKey,
+            },
+            redirect: 'follow',
+            referrerPolicy: 'no-referrer',
+            body: JSON.stringify(data),
+        });
+
+        return await response.json();
+    }
+
+    private async _convert_token(testMode: boolean, checkoutcomkey: string): Promise<PaymentMethodData> {
+        const payment = this._getPayment();
+        let token: CheckoutcomGooglePayToken | undefined;
+        try {
+            token = JSON.parse(payment.paymentData.nonce);
+        } catch (err) {
+            throw new InvalidArgumentError('Unable to parse response from GooglePay.');
+        }
+
+        if ( token && token.protocolVersion ) {
+
+            return this.requestCheckoutcomTokenize(testMode, checkoutcomkey, {
+                type: 'googlepay',
+                token_data: token,
+            })
+                .then(checkoutcomToken => {
+                    if (!checkoutcomToken || !checkoutcomToken.token) {
+                        throw new InvalidArgumentError('Unable to parse response from Checkout.com');
+                    }
+                    payment.paymentData.nonce = checkoutcomToken.token;
```

Comment:
I suggest you not to mutate the object because all objects within this library are considered to be immutable. So doing that can lead to unexpected behaviours (as other parts of the code assume objects don't change). Here, I would return a new object instead. i.e.:

```ts
return {
    ...payment,
    paymentData: {
        ...payment.paymentData,
        nonce: checkoutcomToken.token,
    },
};
```

Reply:
changed as suggested